cli/src/pcluster/models/cluster.py

```diff
@@ -510,69 +555,165 @@ def head_node_ip(self):
             raise ClusterActionError("Head node: {0}\nCannot get ip address.".format(head_node.state.upper()))
         return ip_address
 
-    def start(self):
-        """Start the cluster."""
-        scheduler = self.config.scheduling.scheduler
-        if scheduler == "awsbatch":
-            LOGGER.info("Enabling AWS Batch compute environment : %s", self.name)
-            try:
-                compute_resource = self.config.scheduling.queues[0].compute_resources[0]
-
-                AWSApi.instance().batch.enable_compute_environment(
+    def has_running_capacity(self, updated_value: bool = False):
+        """Return True if the cluster has running capacity. Note: the value will be cached."""
+        if self.__has_running_capacity is None or updated_value:
+            if self.stack.scheduler == "slurm":
+                self.__has_running_capacity = (
+                    ComputeFleetStatusManager(self.name).get_status() != ComputeFleetStatus.STOPPED
+                )
+            elif self.stack.scheduler == "awsbatch":
+                self.__has_running_capacity = self.get_running_capacity() > 0
+        return self.__has_running_capacity
+
+    def get_running_capacity(self, updated_value: bool = False):
+        """Return the number of instances or desired capacity. Note: the value will be cached."""
+        if self.__running_capacity is None or updated_value:
+            if self.stack.scheduler == "slurm":
+                self.__running_capacity = len(self.compute_instances)
+            elif self.stack.scheduler == "awsbatch":
+                self.__running_capacity = AWSApi.instance().batch.get_compute_environment_capacity(
                     ce_name=self.stack.batch_compute_environment,
-                    min_vcpus=compute_resource.min_vcpus,
-                    max_vcpus=compute_resource.max_vcpus,
-                    desired_vcpus=compute_resource.desired_vcpus,
                 )
-            except Exception as e:
-                raise ClusterActionError(f"Unable to enable Batch compute environment. {str(e)}")
+        return self.__running_capacity
 
-        else:  # scheduler == "slurm"
-            stack_status = self.stack.status
-            if "IN_PROGRESS" in stack_status:
-                raise ClusterActionError(f"Cannot start compute fleet while stack is in {stack_status} status.")
-            if "FAILED" in stack_status:
-                LOGGER.warning("Cluster stack is in %s status. This operation might fail.", stack_status)
+    def start(self):
+        """Start the cluster."""
+        try:
+            scheduler = self.config.scheduling.scheduler
+            if scheduler == "awsbatch":
```

Comment:
Shouldn't this be delefated to a AwsBatchCluster instance?

Reply:
Yes, it's in the TODO list.